pkg/kubelet/cm/types.go

```diff
@@ -55,6 +55,7 @@ type CgroupConfig struct {
 	ResourceParameters *ResourceConfig
 }
 
+//go:generate mockgen -source=types.go -destination=testing/cgroupmanager_mock.go -package=testing CgroupManager
```

Comment:
Hi @ehashman, I have one question about the destination of generated mock file.

The  "import cycle" issue occurs me when I try to use the generated `MockCgroupManager` from package `pkg/kubelet/cm/testing`;

I am trying to test `podContainerManagerImpl`, method `Destory()`, I need to use `MockCgroupManager` to stub the `Destroy` method. I write the tests in file "pkg/kubelet/cm/container_manager_linux_test.go", it would leds to import cycle between "" and "package k8s.io/kubernetes/pkg/kubelet/cm", "imports k8s.io/kubernetes/pkg/kubelet/cm/testing"

Maybe we should put the generated mock file into "cgroupmanager_mock.go" instead of "testing/cgroupmanager_mock.go" ?

Reply:
Hm I missed that - yeah we probably need to not have a testing package, because any other change would worsen the CM API